proof/refine/InvariantUpdates_H.thy

```diff
@@ -0,0 +1,457 @@
+(*
+ * Copyright 2021, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ *)
+
+theory InvariantUpdates_H
+imports ArchInvsLemmas_H
+begin
+
+(* generic consequences which require arch-specific proofs from ArchInvsLemmas_H *)
+arch_requalify_facts
+  cte_wp_atE'
+  cte_wp_at_cteI'
+  tcb_at_cte_at'
+  typ_at_lift_valid_cap'
+  valid_arch_tcb_lift'
+  tcb_at_cte_at'
+  gen_objBitsT_simps
+  objBitsT_koTypeOf (* FIXME arch-split: consider declaring [simp] here rather than Schedule_R *)
```

Comment:
That might be nice to do, but I'd leave it for later, so that the fallout from `[simp]` isn't mixed in with the rest. I.e. leave in the FIXME for now.

Reply:
Agreed, I think it'll also be easier to fix the fallout when things are more arch-split. Currently having to go do a fix 5 times is annoying.